parastell/parastell.py

```diff
@@ -47,251 +48,315 @@ class Stellarator(object):
 
     Arguments:
         vmec_file (str): path to plasma equilibrium VMEC file.
-        logger (object): logger object (defaults to None). If no logger is
-            supplied, a default logger will be instantiated.
+        logger (object): logger object (optional, defaults to None). If no
+            logger is supplied, a default logger will be instantiated.
     """
 
     def __init__(
-            self,
-            vmec_file,
-            logger=None
+        self,
+        vmec_file,
+        logger=None
     ):
-        self.vmec_file = vmec_file
-
+        
         self.logger = logger
-        if self.logger == None or not self.logger.hasHandlers():
-            self.logger = log.init()
-
-        self.vmec = read_vmec.VMECData(self.vmec_file)
+        self.vmec_file = vmec_file
 
         self.invessel_build = None
         self.magnet_set = None
         self.source_mesh = None
 
-    def construct_invessel_build(self, invessel_build):
+    @property
+    def vmec_file(self):
+        return self._vmec_file
+    
+    @vmec_file.setter
+    def vmec_file(self, file):
+        self._vmec_file = file
+        self._vmec_obj = read_vmec.VMECData(self._vmec_file)
+        if Path(self._vmec_file).suffix != '.nc':
+            e = ValueError(
+                'Plasma equilibrium VMEC data file input has extension '
+                f'"{Path(self._vmec_file).suffix}". File format must be '
+                'netCDF (".nc").'
+            )
+            self._logger.error(e.args[0])
+            raise e
```

Comment:
This could be limiting... I'm not sure we want to be this heavy-handed with file extensions.

Reply:
Won't `read_vmec` complain for this anyway, if the file is the wrong format?

Reply:
It doesn't seem like `read_vmec` has an exception built-in for this, but I imagine netCDF4 might. The file gets read as
```python
from netCDF4 import Dataset
...
self.data = Dataset(fname)
```
where `fname` is the VMEC file name.